### PR TITLE
Support ACS for Language connector

### PR DESCRIPTION
**Pull request by LucGenetier**
Add new connector support: Azure Cognitive Service for Language
Focus on Conversations (CLU) (2022-05-01) API

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Connectors/ArgumentMapper.cs [hunk 1]
```diff
@@ -209,6 +268,13 @@ public ArgumentMapper(IEnumerable<OpenApiParameter> parameters, OpenApiOperation
                 {
                     map.Add(parameterName, value);
                 }
+                else
+                {
+                    if (value is RecordValue r)
```
Comment (LucGenetier) [inline src/libraries/Microsoft.PowerFx.Connectors/ArgumentMapper.cs:273]: will cleanup

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.IsInternal(Microsoft.OpenApi.Models.OpenApiParameter)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.ToFormulaType(Microsoft.OpenApi.Models.OpenApiSchema)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.TryGetDefaultValue(Microsoft.OpenApi.Models.OpenApiSchema,System.String@)

File: src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs [hunk 2]
```diff
@@ -95,41 +96,68 @@ public static string[] GetOptions(this OpenApiParameter param)
...
         {
+            // https://learn.microsoft.com/en-us/connectors/custom-connectors/openapi-extensions#x-ms-trigger
+            // Identifies whether the current operation is a trigger that produces a single event.
+            // The absence of this field means this is an action operation.
             var isTrigger = op.Extensions.ContainsKey("x-ms-trigger");
             return isTrigger;
         }
 
-        public static bool TryGetDefaultValue(this OpenApiSchema schema, out string defaultValue)
+        public static bool TryGetDefaultValue(this OpenApiSchema schema, FormulaType formulaType, out FormulaValue defaultValue)
```
Comment (LucGenetier) [inline src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs:106]: >FormulaValue [](http://example.com/codeflow?start=102&length=12)

Key change: we don't just return a string but now it's a real FormulaType

File: src/tests/Microsoft.PowerFx.Connectors.Tests/Microsoft.PowerFx.Connectors.Tests.csproj [hunk 3]
```diff
@@ -28,6 +28,11 @@
     <EmbeddedResource Include="Responses\*.json" />
   </ItemGroup>
 
+  <ItemGroup>
+    <None Remove="Responses\Azure Cognitive Service for Language_Response.json" />
+    <None Remove="Swagger\Azure Cognitive Service for Language.json" />
+  </ItemGroup>
```
Comment (LucGenetier) [inline src/tests/Microsoft.PowerFx.Connectors.Tests/Microsoft.PowerFx.Connectors.Tests.csproj:34]: to be removed

File: src/libraries/Microsoft.PowerFx.Connectors/ConnectorFunction.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+using Microsoft.AppMagic.Authoring.Texl.Builtins;
+using Microsoft.OpenApi.Models;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Connectors
+{
+    public class ConnectorFunction
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/ConnectorFunction.cs:21]: >ConnectorFunction [](http://example.com/codeflow?start=17&length=17)

Can we minimize the public surface here as much as possible?

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.IsInternal(Microsoft.OpenApi.Models.OpenApiParameter)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.ToFormulaType(Microsoft.OpenApi.Models.OpenApiSchema)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.TryGetDefaultValue(Microsoft.OpenApi.Models.OpenApiSchema,System.String@)

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.IsInternal(Microsoft.OpenApi.Models.OpenApiParameter)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.ToFormulaType(Microsoft.OpenApi.Models.OpenApiSchema)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.TryGetDefaultValue(Microsoft.OpenApi.Models.OpenApiSchema,System.String@)

**Review by MikeStall — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.IsInternal(Microsoft.OpenApi.Models.OpenApiParameter)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.ToFormulaType(Microsoft.OpenApi.Models.OpenApiSchema)<br />❌CP0002 M:Microsoft.PowerFx.Connectors.OpenApiExtensions.TryGetDefaultValue(Microsoft.OpenApi.Models.OpenApiSchema,System.String@)

**Pull request merged**